File: go/singleton/Logger.go
package main

import (
	"fmt"
	"sync"
)

type Logger struct{}

var instance *Logger
var once sync.Once

func GetInstance() *Logger {
	once.Do(func() {
		instance = &Logger{}
	})
	return instance
}

func (l *Logger) Log(message string) {
	fmt.Printf("[LOG]: %s\n", message)
}

File: python/singleton/Logger.py
class Logger:
    _instance = None  # instancia única

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(Logger, cls).__new__(cls)
        return cls._instance

    def log(self, message: str) -> None:
        print(f"[LOG]: {message}")
